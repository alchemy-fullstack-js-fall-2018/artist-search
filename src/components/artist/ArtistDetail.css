.artist {
  background-color: rgb(164, 212, 229);
  font-size: 20px;
  font-family: Arial, Helvetica, sans-serif;
}

li {
  margin: 30px;
  font-size: 25px;
}
