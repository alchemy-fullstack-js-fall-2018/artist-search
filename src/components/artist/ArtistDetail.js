import React, { Component } from 'react';
import { getArtist } from '../../services/musicSearch';
// import styles from './ArtistDetail.css';
import Songs from '../songs/Songs';

export default class ArtistDetail extends Component {
  state = {
    artist: {}
  };

  fetchArtist = () => {
    getArtist(this.props.match.params.id)
      .then(artist => this.setState({ artist }));
  };

  componentDidMount() {
    this.fetchArtist();
  }

  render() {
    const { id, name, songs } = this.state.artist;
    if(!songs) return null;

    return (
      // <div className={styles.artist}>
      <div>
        <h3>{name}</h3>
        <h4>BAND ID: {id}</h4>
        <Songs artist={name} songs={songs || 'Loading...'} />
      </div>
    );
  }
}
