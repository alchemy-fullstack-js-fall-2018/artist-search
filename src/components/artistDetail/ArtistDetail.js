import React, { Component, Fragment } from 'react';
import { getArtist } from '../../services/api';
import PropTypes from 'prop-types';
import Song from '../song/Song';

export default class Artist extends Component {
    static propTypes = {
        match: PropTypes.object.isRequired,
    };

    state = {
        artist: null
    };

    fetchArtist = () => {
        getArtist(this.props.match.params.id)
            .then(artist => this.setState({ artist }));
    };

    componentDidMount() {
        this.fetchArtist();
    }

    render() {
        if(!this.state.artist) return null;
        const { name, works } = this.state.artist;
        
        const songs = works.map(song => {
            return (
                <Song key={song.id}
                    artistName={name} 
                    songName={song.title} />
            );
        });

        return (
            <Fragment>
                <h2>Artist Detail for {name}</h2>
                {songs}
            </Fragment>
        );
    }
}
