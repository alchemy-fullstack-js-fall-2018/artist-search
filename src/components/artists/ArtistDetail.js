import React, { Component } from 'react';
import WorkListItem from '../works/WorkListItem';
import { getArtist } from '../../services/fetch';

export default class ArtistDetail extends Component {
  state = {
    artist: {}
  };

  fetchArtist = () => {
    getArtist(this.props.match.params.id)
      .then(artist => this.setState({ artist }));
  };

  componentDidMount() {
    this.fetchArtist();
  }

  render() {
    const { name, works = [] } = this.state.artist;
    return (
      <div>
        <h2>{name}</h2>
        <p>{works.map(work => <WorkListItem key={work.id} artist={name} title={work.title} />)}</p>
      </div>
    );
  }
}
