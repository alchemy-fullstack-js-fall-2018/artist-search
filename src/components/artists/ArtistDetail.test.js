import React from 'react';
import { shallow } from 'enzyme';
import ArtistDetail from './ArtistDetail';

jest.mock('../../services/fetch', () => {
  const getArtistResult = {
    artist: {
      name: 'Joni Mitchell',
      works: [
        { id: 'abc', title: 'Song1' },
        { id: '123', title: 'Song2' }]
    }
  };
  return { getArtist: () => Promise.resolve(getArtistResult) };
});

describe ('tests for ArtistDetail component', () => {

  it('renders an ArtistDetail component', () => {
    const wrapper = shallow(<ArtistDetail match={ { params: { id: '123' } }} />);
    expect(wrapper).toMatchSnapshot();
  });

});
