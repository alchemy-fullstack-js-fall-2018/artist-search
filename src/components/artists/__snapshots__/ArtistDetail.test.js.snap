// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for ArtistDetail component renders an ArtistDetail component 1`] = `
<div>
  <h2 />
  <p />
</div>
`;
